dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -439,74 +445,220 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
-            contentletLang1.setStringProperty("firstName", "catherine");
-            contentletLang1 = contentletAPI.checkin(contentletLang1, systemUser, false);
-
-            ContentletDataGen.publish(contentletLang1);
-
-            Contentlet contentletLang2 = contentletAPI
-                    .find(contentletLang1.getInode(), systemUser, false);
-            contentletLang2.setInode("");
-            contentletLang2.setLanguageId(lang2.getId());
-            contentletLang2.setStringProperty("firstName", "catalina");
-            contentletLang2 = contentletAPI.checkin(contentletLang2, systemUser, false);
-
-            ContentletDataGen.publish(contentletLang2);
-
-            final Container container = new ContainerDataGen().withContentType(contentletLang1
-                    .getContentType(), "$!{firstName}").nextPersisted();
-
-            ContainerDataGen.publish(container);
+            Contentlet contentletDefaultLang = null;
+            Contentlet contentSecondLang = null;
 
-            final String uuid = UUIDGenerator.generateUuid();
+            if(testCase.createContentInDefaultLanguage) {
+                contentletDefaultLang = createAndPublishEmployeeContent(site, defaultLang, "catherine");
 
-            final Template template = new TemplateDataGen()
-                    .withContainer(container.getIdentifier(), uuid)
-                    .nextPersisted();
+                if(testCase.createContentInSecondLanguage) {
+                    createNewVersionAndPublishExistingEmployeeContent(secondLang, contentletDefaultLang,
+                            "catalina");
+                }
+            } else if(testCase.createContentInSecondLanguage) {
+                contentSecondLang = createAndPublishEmployeeContent(site, secondLang, "catalina");
 
-            TemplateDataGen.publish(template);
-
-            HTMLPageAsset page = new HTMLPageDataGen(folder, template).languageId(lang1.getId())
-                    .nextPersisted();
-
-            HTMLPageDataGen.publish(page);
-
-            final MultiTree multiTree = new MultiTree(page.getIdentifier(),
-                    container.getIdentifier(),
-                    contentletLang1.getIdentifier(), getDotParserContainerUUID(uuid), 0);
-
-            APILocator.getMultiTreeAPI().saveMultiTree(multiTree);
+                if(testCase.createContentInThirdLanguage) {
+                    createNewVersionAndPublishExistingEmployeeContent(thirdLang, contentSecondLang,
+                            "caterina");
+                }
+            }
 
-            final String html = APILocator.getHTMLPageAssetAPI().getHTML(page.getURI(), site, true,
-                    contentletLang1.getIdentifier(), APILocator.systemUser(),
-                    contentletLang1.getLanguageId(), USER_AGENT_DOTCMS_SITESEARCH);
+            Contentlet contentToPassToPage = contentletDefaultLang!=null
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'contentToPassToPage' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)